[AIRFLOW-2205] Remove unsupported args from JdbcHook doc

sekikn opened the pull request:
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2205


### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

JdbcHook's docstring has unsupported arguments,
a wrongly named argument and unimplemented feature description.
This PR fixes them and adds JdbcHook to the API reference.


### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

This PR doesn't need testing since it's documentation fix. I ran docs/build.sh locally and confirmed that the API document was generated as expected.


### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

jgao54 commented on airflow/hooks/jdbc_hook.py:
```diff
@@ -23,21 +23,12 @@ class JdbcHook(DbApiHook):
-    :param sql: the sql code to be executed
-    :type sql: string or string pointing to a template file. File must have
-        a '.sql' extensions.
+    :param jdbc_conn_id: reference to a predefined database
```
can be removed too right?

sekikn replied:
Thanks for the review @jgao54! As you know, DbApiHook and its subclasses take a parameter which name is defined by `conn_name_attr` class variable, so I added it to the docstring. But I've just noticed that other hooks derived from DbApiHook (such as MsSql, MySql, Postgres) don't mention that in their docstring, so I'll remove those lines for consistency with them.

jgao54 replied:
thanks!

sekikn committed:
[AIRFLOW-2205] Remove unsupported args from JdbcHook doc

JdbcHook's docstring has unsupported arguments
and unimplemented feature description.
This PR fixes them and adds JdbcHook to the API reference.
